chain/network/src/routing/actor.rs

```diff
@@ -0,0 +1,319 @@
+use crate::private_actix::{StopMsg, ValidateEdgeList};
+use crate::routing;
+use crate::routing::edge_validator_actor::EdgeValidatorActor;
+use crate::stats::metrics;
+use crate::store;
+use actix::{
+    ActorContext as _, ActorFutureExt, Addr, Context, ContextFutureSpawner as _, Running,
+    WrapFuture as _,
+};
+use near_network_primitives::time;
+use near_network_primitives::types::Edge;
+use near_performance_metrics_macros::perf;
+use near_primitives::network::PeerId;
+use near_rate_limiter::{ActixMessageResponse, ActixMessageWrapper, ThrottleToken};
+use parking_lot::RwLock;
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+use tracing::warn;
+
+/// Actor that maintains routing table information.
+///
+/// We store the following information
+///   - routing information (where a message should be send to reach given peer)
+///
+/// We use store for following reasons:
+///   - store removed edges to disk
+///   - we currently don't store active edges to disk
+pub(crate) struct Actor {
+    clock: time::Clock,
+    my_peer_id: PeerId,
+    // Graph is locked for writes only by this Actor, and only to update a bunch of edges.
```

Comment:
so this actor is the only person that can update this graph ? (as it wasn't clear from the comment)

Reply:
yes, in fact recently I realized that once I get rid of all IBF-related testonly methods, graph will become an implementation detail of routing::Actor, so this reference will go away soon.

Reply:
updated the comment.